receiver/splunkhecreceiver/receiver.go

```diff
@@ -391,9 +392,20 @@ func (r *splunkReceiver) handleRawReq(resp http.ResponseWriter, req *http.Reques
 	}
 }
 
-func (r *splunkReceiver) extractChannelHeader(req *http.Request) (string, bool) {
-	if headers, ok := req.Header[splunk.HTTPSplunkChannelHeader]; ok {
-		return headers[0], true
+func (r *splunkReceiver) extractChannel(req *http.Request) (string, bool) {
+	// check header
+	for k, v := range req.Header {
+		if strings.EqualFold(k, splunk.HTTPSplunkChannelHeader) {
+			return strings.ToUpper(v[0]), true
+		}
+	}
+	// check query param
+	u, _ := url.Parse(req.URL.String())
```

Comment:
`req.URL` must be already parsed. I don't think you need to render and parse it back